pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -832,3 +836,44 @@ Iterable<ArchiveIssue> checkTopics(String pubspecContent) sync* {
     }
   }
 }
+
+// Map from allowed hook-files to minimum SDK lower bound constraint
+final _allowedHookFiles = <String, Version>{
+  'hook/build.dart': Version.parse('3.6.0-0'),
+  'hook/link.dart': Version.parse('3.6.0-0'),
+};
+
+Version? _minVersion(VersionConstraint? constraint) {
+  if (constraint == null) {
+    return null;
+  }
+  if (constraint is Version) {
+    return constraint;
+  }
+  if (constraint is VersionRange) {
+    return constraint.min;
+  }
+  return null;
+}
+
+Iterable<ArchiveIssue> checkHooks(
+  Version? minimumSdkConstraint,
+  Set<String> fileNames,
+) sync* {
+  final hookFiles = fileNames.where((n) {
+    final parts = p.split(n);
+    return parts.length > 1 && parts.first == 'hook';
+  }).toList();
```

Comment:
Have we considered:
```suggestion
  final hookFiles = fileNames.where((f) => f.startsWith('hook/'));
```

We don't even need to use `toList()` :rofl: